packages/hop-node/src/watchers/BondWithdrawalWatcher.ts

```diff
@@ -129,21 +150,16 @@ class BondWithdrawalWatcher extends BaseWatcher {
       }
       availableCredit = availableCredit.sub(pendingAmounts).sub(amount)
     }
+
+    this.lastAvailableCredit[sourceChain][destinationChain] = availableCredit
```

Comment:
I see `lastAvailableCredit` is only updated when a bond withdrawal is attempted; do you think it should also be updated when credit or debit is updated in general which have an effect on available liquidity (e.g. staking, settling, challenges)?

Reply:
That is a great thought & I didn't originally consider it.

My vote would be to not update in those cases & leave it as-is because I believe that it can introduce some edge cases while providing a negligible benefit. By negligible benefit, I mean that I see the benefit of adding the state updates in other watchers as being a user who is waiting will have their tx bonded faster (seconds or minutes). I believe, based on our current usage, this will be negligible because other user's send txs every few mins that will update this value anyway.

If we were to add the updates everywhere, one complex case I am imagining is "race conditions". Imagine we have two watchers who update it w/in milliseconds of each other. Imagine, in this case, that the one who updates it last actually did the `amount` calcs 1s before, which means it is overwriting w/ bad data. I think this can almost be guaranteed to not happen if we do the calc on the line before the state update, but then I _think_ we need to introduce the `includePendingAmount` in all watchers too.

Open for discussion if you think differently!

Reply:
I don't think there would have to be updates everywhere; I think the available liquidity logic would be in a single place and the update gets triggered after every resync (it can be in the postSyncHandler). This makes it so that the available liquidity amount is always up to date after receiving new events

Reply:
I like that. I'll take a whack at it and send over a commit with this change.